nautobot/dcim/views.py

```diff
@@ -2492,34 +2493,50 @@ def extra_context(self):
-    queryset = Interface.objects.filter(
-        # Avoid duplicate connections by only selecting the lower PK in a connected pair
-        _path__isnull=False,
-        pk__lt=F("_path__destination_id"),
-    ).order_by("device")
+    queryset = (
+        Interface.objects.filter(_path__isnull=False)
+        .exclude(
+            # If an Interface is connected to another Interface, avoid returning both (A, B) and (B, A)
+            # Unfortunately we can't use something consistent to pick which pair to exclude (such as device or name)
+            # as _path.destination is a GenericForeignKey without a corresponding GenericRelation and so cannot be
+            # used for reverse querying.
+            # The below at least ensures uniqueness, but doesn't guarantee whether we get (A, B) or (B, A);
+            # we fix it up to be consistently (A, B) in queryset_to_csv() below.
+            _path__destination_type=ContentType.objects.get_for_model(Interface),
+            pk__lt=F("_path__destination_id"),
+        )
+        .order_by("device__identifier", "name")
+    )
     filterset = filters.InterfaceConnectionFilterSet
     filterset_form = forms.InterfaceConnectionFilterForm
     table = tables.InterfaceConnectionTable
     template_name = "dcim/connections_list.html"
 
     def queryset_to_csv(self):
-        csv_data = [
-            # Headers
-            ",".join(["device_a", "interface_a", "device_b", "interface_b", "reachable"])
-        ]
+        headers = ",".join(["device_a", "interface_a", "device_b", "interface_b", "reachable"])
+        csv_data = []
         for obj in self.queryset:
-            csv = csv_format(
-                [
-                    obj._path.destination.device.identifier if obj._path.destination else None,
-                    obj._path.destination.name if obj._path.destination else None,
-                    obj.device.identifier,
-                    obj.name,
-                    obj._path.is_active,
-                ]
-            )
+            obj_data = [
+                obj.device.identifier,
+                obj.name,
+                None,  # see below
+                None,  # see below
+                str(obj._path.is_active),
+            ]
+            if obj._path.destination and hasattr(obj._path.destination, "device"):
+                obj_data[2] = obj._path.destination.device.identifier
+            if obj._path.destination and hasattr(obj._path.destination, "name"):
+                obj_data[3] = obj._path.destination.name
+            # Ensure consistent ordering (see comment on queryset above) for the CSV export at least
+            if obj_data[2] and (
+                obj_data[0] > obj_data[2] or (obj_data[0] == obj_data[2] and obj_data[1] > obj_data[3])
+            ):
+                # Swap a and b!
+                obj_data = [*obj_data[2:4], *obj_data[0:2], *obj_data[4:]]
+            csv = csv_format(obj_data)
```

Comment:
This is looking a bit arcane. Can you please comment this up?

Also this might be easier to read and maintain if `obj_data` were an `OrderedDcit` instead of a `list`? Could still pass the values by reference to `csv_format()`?

Reply:
I'll see what I can do to make this more readable. It's kinda gross admittedly.

Reply:
A `namedtuple` for the comparisons actually might be the sweet spot. Since you can reference the fields by name, and then they still behave like a tuple.